### PR TITLE
INTYGFV-15549: Refactored AESEncrypter.

### PR DESCRIPTION
**Pull request by ChrilleSigerhed**

### CODE REVIEW

**Review by mhornfeldt — APPROVED** [review]

File: web/src/test/java/se/inera/intyg/rehabstod/web/controller/api/SjukfallControllerTest.java [hunk 1]
```diff
@@ -121,6 +122,9 @@ public class SjukfallControllerTest {
     @InjectMocks
     private SjukfallController sjukfallController = new SjukfallController();
 
+    @Mock
+    private AESEncrypter aesEncrypter;
```
Comment (mww-sopra-steria) [inline web/src/test/java/se/inera/intyg/rehabstod/web/controller/api/SjukfallControllerTest.java:126]: Add test for the encrypted and non-encrypted patientId

File: web/src/main/java/se/inera/intyg/rehabstod/service/sjukfall/util/AESEncrypter.java [hunk 2]
```diff
@@ -24,44 +24,49 @@
...
 public class AESEncrypter {
 
-    private static final String KEY = "0123456789abcdef";
+    private static final Logger LOG = LoggerFactory.getLogger(AESEncrypter.class);
     private static final String INIT_VECTOR = "fedcba9876543210";
+    @Value("${aes.encryption.key}")
+    private String key;
 
-    public static String encryptPatientId(String value) {
+    public String encryptPatientId(String value) {
```
Comment (mww-sopra-steria) [inline web/src/main/java/se/inera/intyg/rehabstod/service/sjukfall/util/AESEncrypter.java:40]: Maybe could refactor the class to reuse the cipher object. One for decrypt and one for encrypt. 

Also consider the name of the class to something more specific. 

Would be great to reintroduce some unit tests. You can use the @Value in the constructor and initialise it for test with different keys.

**Review by mww-sopra-steria — CHANGES_REQUESTED** [review]
A few comments that we discussed separately.

**Review by mww-sopra-steria — APPROVED** [review]
Looks good!

**Pull request merged**